www/js/useAppConfig.ts

```diff
@@ -0,0 +1,24 @@
+import { useEffect, useState } from "react";
+import { getAngularService } from "./angular-react-helper"
+
+const useAppConfig = () => {
+
+  const [appConfig, setAppConfig] = useState<any>({});
+  const [loading, setLoading] = useState<boolean>(true);
+
+  const DynamicConfig = getAngularService('DynamicConfig');
+  const $ionicPlatform = getAngularService('$ionicPlatform');
+
+  useEffect(() => {
+    $ionicPlatform.ready().then(() => {
+      DynamicConfig.configReady().then((config) => {
+        setAppConfig(config);
+        setLoading(false);
```

Comment:
are we sure we are done with all loading here? Or is this flag only used to track when the app config has loaded? Where do we actually check for this?

Reply:
`loading` is scoped to this hook and only refers to the appConfig not being ready yet.

It's used like this in a few components including TripCard and PlaceCard:
`const { appConfig, loading } = useAppConfig()`

When `loading` changes, we know it's safe to access `appConfig`.